Code/GraphMol/Atom.cpp

```diff
@@ -623,11 +641,10 @@ bool Atom::hasValenceViolation() const {
       std::any_of(bonds.begin(), bonds.end(), is_query)) {
     return false;
   }
-
-  try {
-    calculate_explicit_valence(*this, true);
-    calculate_implicit_valence(*this, true);
-  } catch (const RDKit::AtomValenceException &) {
+  bool strict = false;
+  bool checkIt = true;
+  if (calculateExplicitValence(*this, strict, checkIt) == -1 ||
+      calculateImplicitValence(*this, strict, checkIt) == -1) {
     return true;
   }
   return false;
```

Comment:
is the concern here the performance of the try/catch? It seems odd to me to have both strict *and* checkIt.

Reply:
> is the concern here the performance of the try/catch? It seems odd to me to have both strict _and_ checkIt.

I agree about the oddness, but using try/catch as part of the normal flow control - as far as this function is concerned, an atom with valence violations is not exceptional - feels wrong (as opposed to just odd)